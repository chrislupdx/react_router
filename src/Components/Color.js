import React from 'react';
import PropTypes from 'prop-types';

function Color({ match }) {
  const style = {
    backgroundColor: decodeURIComponent(match.params.color),
    width: '100vw',
    height: '100vh'
  };

  return <div style={style}></div>;
}

Color.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      color: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default Color;
