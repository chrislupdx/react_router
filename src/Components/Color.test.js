import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color', () => {
  it('renders', () => {
    const match = {
      params: {
        color: 'purple'
      }
    };
    const wrapper = shallow(<Color match={match} />);
    expect(wrapper).toMatchSnapshot();
  });
});
