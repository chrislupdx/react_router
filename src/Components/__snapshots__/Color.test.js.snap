// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color renders 1`] = `
<div
  style={
    Object {
      "backgroundColor": "purple",
      "height": "100vh",
      "width": "100vw",
    }
  }
/>
`;
